### DISCUSSION TITLE
To-do for website

### DISCUSSION THREAD

**Discussion by KendallDoesCoding** [collab-community/journey-book#44] [General]
- Before merging website to this repo:

- [ ] Add journeys in the sidebar using Vue.js (assigned to @vikasganiga05)
- [ ] Get a domain in place
       - I can get a .tech domain, etc for free using GitHub Student Plan, but if fine, we can get a freenom domain coz I may want to use the .tech in future.
       - @Adarsh-jaiss Please sign up for the GitHub Student Plan, then we use your .tech domain if that's fine with you

- Before or after merging to this repo:
- [ ]  gh-action to update text (sections) from README.md to the individual page of the website (not so important)
- [ ] gh-action to hyperlink the name of the user from .json file and link their journey via a hyperlink in the sidebar
- [ ] Maybe add a navbar.

@vikasganiga05 As your the docsify expert, Will require your help to merge the website to this repo once done with the above,

Thanks,
KendallDoesCoding

**Comment by vikasganiga05** [comment]
Hi @KendallDoesCoding , I will complete the sidebar navigation once you merge the PR to the final branch on the docsify-test repo. I am not a docsify expert :sweat_smile: let me know when you are ready to integrate into this main project.

**Comment by KendallDoesCoding** [comment]
Hey Vikas. The PR is made by the branch "final" so the final website code
is currently in that branch. I only made the PR so we could get the deploy
preview link.

By that I meant, you know more then on docsify.

We will merge the final website once you finish making the sidebar using
Vue.js

Thanks,
KendallDoesCoding

**Reply by vikasganiga05** [reply]
Okay, Then i will start working on it now. I will let you know once I completed it.
When we have lots of users the sidebar will get messy. How we can prevent this do you have any idea?

**Comment by KendallDoesCoding** [comment]
cool! let me know when it's done.


We can either make the journeys section very small with css, if possible,
or keep the sidebar only for journey's and move the other stuff to the
navbar.

**Reply by vikasganiga05** [reply]
Hi, I added the user journey links to the sidebar. There are some issues :point_down: 
- The title is not changing to the user's name.
- The journey section in the sidebar is refreshing every time when sidebar links are clicked.

Check it out &rarr; https://deploy-preview-2--neontuts-docsify-test.netlify.app/

**Reply by vikasganiga05** [reply]
And docsify doesn't support Vue.js on the sidebar. It took too long to figure out the solution. But the better solution is to use github action to create those links on the sidebar.

**Comment by KendallDoesCoding** [comment]
For now it works, but we may have to find a gh-action.

Just realised, the sidebar doesnt work on mobile though, what should we do,
any ideas? Coz on the sidebar all the pages are listed.

**Reply by vikasganiga05** [reply]
The sidebar is working on mobile it's toggled off. The sidebar toggle button is hidden under the Netlify overlay.

**Reply by KendallDoesCoding** [reply]
Oh ok, makes sense.

**Comment by KendallDoesCoding** [comment]
@Adarsh-jaiss Please confirm if you like this website - https://deploy-preview-2--neontuts-docsify-test.netlify.app/

If you like the website @Adarsh-jaiss @vikasganiga05 , @vikasganiga05 Please create a PR to merge the website into this repo, you can make a branch in this repo and add the website code. Then, we'll talk about the domain, etc.

**Reply by KendallDoesCoding** [reply]
I like the website!

**Reply by vikasganiga05** [reply]
Yeah, I like it too. I created the GitHub Action for the sidebar links. I will add that to the test repo.
GitHub Action &rarr; https://github.com/neontuts/gh-actions-markdown-link-generator

**Reply by vikasganiga05** [reply]
@KendallDoesCoding I modified this &rarr; https://github.com/marketplace/actions/json2table github-action. Which you shared a while ago :slightly_smiling_face:

**Comment by KendallDoesCoding** [comment]
Perfect! Thanks a ton!

**Reply by vikasganiga05** [reply]
You're welcome. I added the dark and light toggle. Check it out &rarr; https://deploy-preview-2--neontuts-docsify-test.netlify.app/

**Comment by KendallDoesCoding** [comment]
For me, my chrome only is messed up, everything is darkmode for some
reason. ,Adarsh and Vikas please confirm if it's working properly.

**Reply by vikasganiga05** [reply]
It's working fine on the Firefox browser. :point_down: 

![image](https://user-images.githubusercontent.com/74750414/179397129-6e677547-6386-4f6b-8ed4-0136f20e4995.png)

![image](https://user-images.githubusercontent.com/74750414/179397134-aeab230a-ad73-4a5c-b54d-c0b36fb2299b.png)

**Comment by KendallDoesCoding** [comment]
Oh nice, glad it was useful.

**Comment by KendallDoesCoding** [comment]
Ok cool.

Please merge the website to this repo

**Reply by vikasganiga05** [reply]
Done :+1: Please review it.

**Comment by vikasganiga05** [comment]
Hi @Adarsh-jaiss & @KendallDoesCoding , I added the website code from the test repo to **[docsify](https://github.com/collab-community/journey-book/tree/docsify)** branch. Here is the PR #47 . Please take a look at it. If anything is wrong please let me know.

Thank You :pray:

**Comment by KendallDoesCoding** [comment]
Hi Vikas.

Thanks a ton. Will review later in the day

**Reply by vikasganiga05** [reply]
Your welcome :slightly_smiling_face:

**Comment by KendallDoesCoding** [comment]
@vikasganiga05 The website looks amazing, thanks for all the work you put into it!
